adapter/reaction_event_repository.py

```diff
@@ -51,3 +55,34 @@ def create_reaction_event(self, event_create: ReactionEventCreate):
                 )
                 session.add(new_reaction_event)
                 session.commit()
+    
+    def delete_reaction_event(
+        self,
+        reaction: Reaction,
+        source_adapter_message_id: str,
+        source_adapter_user_id: str
+    ):
+        self.logger.bind(
+            reaction=reaction,
+            message_id=source_adapter_message_id,
+            user_id=source_adapter_user_id,
+        ).info("deleting reaction event")
+
+        with self.create_session() as session:
+            with self.logger.catch(message="delete reaction event error", reraise=True):
+                reaction_event = (
+                    session.query(ReactionEventModel)
+                    .filter_by(
+                        reaction=reaction,
+                        source_adapter_message_id=source_adapter_message_id,
+                        source_adapter_user_id=source_adapter_user_id,
+                    )
+                    .first()
+                )
+                
+                if reaction_event:
+                    session.delete(reaction_event)
+                    session.commit()
+                    self.logger.info("Reaction event deleted successfully")
+                else:
+                    self.logger.warning("No matching reaction event found for deletion")

```

Comment:
There shouldn't be any need to check if the event exist or not no? If an object doesn't exist, then the operation should still succeed since it shouldn't really matter if it's trying to delete something that doesn't exist.

The concept is called [idempotency](https://blog.algomaster.io/p/idempotency-in-distributed-systems) and it's often applied in both monolithic and distributed systems.

The benefit is that we reduce the amount of queries taken and reduces the processing time as well